Python/bytecodes.c

```diff
@@ -993,7 +1023,7 @@ dummy_func(
             // GH-99729: We need to unlink the frame *before* clearing it:
             _PyInterpreterFrame *dying = frame;
             frame = tstate->current_frame = dying->previous;
-            _PyEval_FrameClearAndPop(tstate, dying);
+            _PyEval_ClearGenFrame(tstate, dying);
```

Comment:
Is the only difference between `RETURN_VALUE_FUNC` and `RETURN_VALUE_GEN` the call to `_PyEval_ClearThreadFrame` vs `_PyEval_ClearGenFrame`?   Couldn't `_PyEval_FrameClearAndPop` just switch on whether or not `frame->owner == FRAME_OWNED_BY_GENERATOR`?

Reply:
That's exact what it used to do.

We generally want to make decisions at compile time if we can.
It's the second bullet in https://github.com/python/cpython/issues/125837#issue-2605185061

Reply:
> We generally want to make decisions at compile time if we can.

We also made some efforts to reduce the number of bytecodes and the size of the eval loop implementation.
And copy-pasting 20 lines of code to change one is... controversial.

Reply:
> And copy-pasting 20 lines of code to change one is... controversial.

It's 16 lines, but I see your point.

Doing this gives more flexibility to refactor and optimize the two return paths separately.
Both `_PyEval_ClearGenFrame` and `_PyEval_ClearThreadFrame` call `_PyFrame_ClearExceptCode()` but have to do 
different things around that call.

For example, we could end up with something like this:
```C
macro(RETURN_VALUE_FUNC) =  _CLEAR_FRAME + _POP_THREAD_FRAME + _RETURN_VALUE;
macro(RETURN_VALUE_GEN) = _GEN_POP_EXC_STATE + _CLEAR_FRAME + _RETURN_VALUE;
```
but not in this PR.